Automattic/newspack-plugin
# Support: add Happychat

### All Submissions:






### Changes proposed in this Pull Request:



This introduces a Chat tab to the Support view:

![image](https://user-images.githubusercontent.com/7383192/75757926-88ea6780-5d33-11ea-9b6f-9b4694216e1d.png)

#### WPCOM authentication whitelist problem

There is one problem with the approach introduced here. 
A WPCOM auth token is needed for authentication to interact with Happychat, so [WPCOM app](https://developer.wordpress.com/apps/) needs to be set up for that. The problem is that such an app has to have a whitelist of request origin and redirect URLs. This list will have to be updated with each new site using the support feature.

#### Design

The chat window is an iframe, so the only way of styling it's contents is via adding new code to the [`happychat-client`](https://github.com/Automattic/happychat-client). Bare in mind that it might take time.

### How to test the changes in this Pull Request:

1. Create [a WPCOM app](https://developer.wordpress.com/apps/) and set:
    1. "Javascript Origins" to your test site URL
    1. "Redirect URLs" to `<your test site URL>/wp-admin/admin.php?page=newspack-support-wizard`
1. Add `NEWSPACK_WPCOM_CLIENT_ID` variable in `wp-config.php`:
```
define('NEWSPACK_WPCOM_CLIENT_ID', '<WPCOM app Client ID>');
```
3. Add ticket-related variables (see #451 )
4. Build the app
5. Visit the support screen, which should be linked to from the Newspack admin menu
6. Observe that there are two tabs - "Submit ticket" & "Chat". Visit the latter.
7. You should see an authentication prompt:
![image](https://user-images.githubusercontent.com/7383192/75759424-d8ca2e00-5d35-11ea-9f72-78fe794b41d7.png)
8. Authenticate. After that you should be taken back to your site, and after a moment of loading, the chat UI should appear.
9. If you send any message, you'll be connected to an HE, so be clear that it's just about testing and not a legitimate support request :) 

### Other information:
